src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelperOfT.cs

```diff
@@ -70,10 +74,11 @@ public override void Contextualize([NotNull] ViewContext viewContext)
         }
 
         /// <inheritdoc />
-        public HtmlString DisplayFor<TValue>([NotNull] Expression<Func<TModel, TValue>> expression,
-                                             string templateName,
-                                             string htmlFieldName,
-                                             object additionalViewData)
+        public HtmlString DisplayFor<TResult>(
+            [NotNull] Expression<Func<TModel, TResult>> expression,
+            string templateName,
+            string htmlFieldName,
```

Comment:
no, I just messed up the context when responding earlier.  `Display[For]()` and `Editor[For]()` helpers have an odd ability to override the `ExpressionHelper.GetExpressionText()` return value.  this is exposed through the separate (and likely little-used) `htmlFieldName` parameter.